extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    internal static class EventGridHelper
+    {
+        internal static ValueTask<ConversionResult> DeserilizeToTargetType(ConverterContext context)
+        {
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Unhandled());
+                }
+                
+                var targetType = context!.TargetType;
+                var item = JsonSerializer.Deserialize(contextSource, targetType);
```

Comment:
I wonder if we should be consistent and use the worker options sterilizer for all

Reply:
I think it might be worth keeping the JsonSerializer for some, since we'd have to convert the contextSource into a byte array and then convert it to a Stream just so we can use the worker options serializer. Using this method makes it much more concise imo.

Reply:
I think we want to use the default JsonSerializer. The EventGridEvent and CloudEvent types have JsonConverters defined that will handle the serialization. If we passed in options that specified separate converters it's possible that those converters would override the standard ones.

Reply:
Sounds good I'll keep the default JsonSerializer then